packages/x-components/src/components/suggestions/base-suggestion.vue

```diff
@@ -32,127 +32,140 @@
-     */
-    @Prop() //TODO: set to true when the suggestions components pass it.
-    protected feature?: QueryFeature;
-
-    /**
-     * The {@link XEvent | XEvents} that will be emitted when selecting a suggestion.
-     *
-     * @public
-     */
-    @Prop({ required: true })
-    protected suggestionSelectedEvents!: Partial<XEventsTypes>;
-
-    /**
-     * Indicates if the curated suggestion should be highlighted.
-     *
-     * @public
-     */
-    @Prop({ default: false, type: Boolean })
-    protected highlightCurated!: boolean;
-
-    /**
-     * The event handler that will be triggered when clicking on a suggestion.
-     *
-     * @remarks
-     * UserAcceptedAQuery: suggestion.query
-     * UserSelectedASuggestion: suggestion
-     * UserClickedAFilter: suggestion.facets[0].filters[0]
-     * Merges the events defined in the suggestionSelectedEvents prop and also emits them
-     *
-     * @returns The {@link XEvent | XEvents} to emit.
-     * @public
-     */
-    protected get events(): Partial<XEventsTypes> {
-      const filterEvent: Partial<XEventsTypes> = this.filter
-        ? { UserClickedAFilter: this.filter }
-        : {};
-      return {
-        UserAcceptedAQuery: this.suggestion.query,
-        UserSelectedASuggestion: this.suggestion,
-        ...this.suggestionSelectedEvents,
-        ...filterEvent
-      };
-    }
-
-    /**
-     * Emits the events when the button is clicked.
-     *
-     * @public
-     */
-    protected emitEvents(): void {
-      forEach(this.events, (event, payload) => {
-        this.$x.emit(event, payload, {
-          target: this.$el as HTMLElement,
-          feature: this.feature
+  export default defineComponent({
+    components: { Highlight },
+    props: {
+      /**
+       * The normalized query of the module using this component.
+       *
+       * @public
+       */
+      query: {
+        type: String,
+        default: ''
+      },
+      /**
+       * The suggestion to render and use in the default slot.
+       *
+       * @public
+       */
+      suggestion: {
+        type: Object as PropType<Suggestion>,
+        required: true
+      },
+      /**
+       * The feature from which the events will be emitted.
+       *
+       * @public
+       */
+      //TODO: set to true when the suggestions components pass it.
+      feature: {
+        type: String as PropType<QueryFeature>,
+        required: false
+      },
+      /**
+       * The {@link XEvent | XEvents} that will be emitted when selecting a suggestion.
+       *
+       * @public
+       */
+      suggestionSelectedEvents: {
+        type: Object as PropType<Partial<XEventsTypes>>,
+        required: true
+      },
+      /**
+       * Indicates if the curated suggestion should be highlighted.
+       *
+       * @public
+       */
+      highlightCurated: {
+        type: Boolean,
+        default: false
+      }
+    },
+    setup(props) {
+      const root = ref<HTMLElement | null>(null);
```

Comment:
Add docs for this.

Reply:
Keep consistency with the decided name of the other PR (either `root` or `el` here)